aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs

```diff
@@ -188,3 +189,21 @@ fn test_aggregator_materialize_overflow() {
     assert_success!(h.run(txn1));
     assert_abort!(h.run(txn2), 131073);
 }
+
+#[test]
+fn test_aggregator_snapshot() {
+    let (mut h, acc) = setup();
+    let txn1 = new(&mut h, &acc, 0, 400);
+    let txn2 = snapshot(&mut h, &acc, 0);
+    let txn3 = snapshot_with_u64_limit(&mut h, &acc, 0);
+    let txn4 = read_snapshot_u128(&mut h, &acc, 0);
+    let txn5 = read_snapshot_u64(&mut h, &acc, 0);
+    let txn6 = try_add_and_read_snapshot_u128(&mut h, &acc, 0, 100);
+
+    assert_success!(h.run(txn1));
```

Comment:
How does this work? If we do not have pieces connected together?

For the future: once we connect end-to-end we should run these txns as a block, not one by one to test